ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/API.hs

```diff
@@ -346,6 +389,10 @@ data ChainDB m blk = ChainDB {
       --
       -- 'False' when the database is closed.
     , isOpen             :: STM m Bool
+
+      -- | Perform a monadic operation holding the read lock on the DB
+      -- changelog.
+    , withLgrReadLock :: forall a. m a -> m a
```

Comment:
One aspect we should document outside this PR is the lock mechanics: why do we need it?